lib/std/fs/file.zig

```diff
@@ -279,12 +279,12 @@ pub const File = struct {
         }
         const times = [2]os.timespec{
             os.timespec{
-                .tv_sec = @divFloor(atime, std.time.ns_per_s),
-                .tv_nsec = @mod(atime, std.time.ns_per_s),
+                .tv_sec = math.cast(isize, @divFloor(atime, std.time.ns_per_s)) catch maxInt(isize),
```

Comment:
So if you tried to set to a date > 2038, you'd instead set it to 2038? that seems wrong...

Reply:
Meh, returning an error is fine as well IMO. I've applied the same approach used in the `sleep` syscall here.

Reply:
Doesn't the `sleep` syscall repeat the max value in a loop? changing the `mtime` of a file can't work like that...

Reply:
No, it just clamps the timestamp to `0xffffffff`.

Reply:
`sleep` is documented this way:

>  Spurious wakeups are possible and no precision of timing is guaranteed. 

which is why clamping is a good solution. It's always a code smell to call sleep. I don't think its behavior is related to this `updateTimes` function.

This is a cross platform abstraction; other platforms allow more bits of data in the timestamps of files. So the question here is, what to do when the number of bits used is too large for the given target? We have the following options:

 * Define the parameter to be "requested time" and clamp to maximum value. Document that the actual time set will be as close to the requested time as possible. This is already kinda how it works because file systems have implementation-defined precision; you may get truncated to seconds, or multiples of seconds, or even to 0 on a poorly implemented FUSE file system.
 * Return an error such as `OutOfRange` or `Overflow`. I think I talked myself out of this one with the previous bullet point.
 * Make the type target-dependent instead of `i64`. This pushes the problem down to the API user. This gives the API user more work to do, and gives up a little bit of abstraction.